providers/softlayer/src/main/java/org/jclouds/softlayer/compute/strategy/SoftLayerComputeServiceAdapter.java

```diff
@@ -447,6 +457,27 @@ public boolean apply(OperatingSystem input) {
               .first();
    }
 
+   private String getNotesFrom(SoftLayerTemplateOptions templateOptions) {
+      if (templateOptions.getNotes().isPresent()) {
+         return templateOptions.getNotes().get();
+      } else {
+         Map<String, String> meta = templateOptions.getUserMetadata();
+         if (meta != null && !meta.isEmpty()) {
+            if (meta.containsKey(USER_META_NOTES)) {
+               return meta.get(USER_META_NOTES);
+            } else {
+               String notes = "User Metadata\n===========\n\n" + Joiner.on("\n").withKeyValueSeparator(": ").join(meta);
+               if (notes.length() > SoftLayerTemplateOptions.NOTES_MAX_LENGTH) {
+                  String suffix = "...\n<truncated>";
+                  notes = notes.substring(0, SoftLayerTemplateOptions.NOTES_MAX_LENGTH - suffix.length()) + suffix;
+               }
+               return notes;
+            }
+         }
+      }
+      return null;
+   }
```

Comment:
Ok. In that case, and given that SoftLayer does not support user metadata I wouldn't have the two approaches of having a dedicated "key" or serialise the entire metadata map. I'd pick just one. We'd document the key to use to set the notes, or we document that we'll assume all user metadata to be serialised as the notes, but not both. Pick the one you prefer :)

Also note that you are not taking not account the notes max length when reading the `USER_META_NOTES` key. Perhaps you also have to truncate it there?